dataprofiler/profilers/profile_builder.py

```diff
@@ -230,7 +236,37 @@ def _update_base_stats(self, base_stats):
             self.null_types, list(base_stats["null_types"].keys())
         )
 
-        for null_type, null_rows in base_stats["null_types"].items():
+        base_min = base_stats["min_id"]
+        base_max = base_stats["max_id"]
+        base_nti = base_stats["null_types"]
+
+        # Check if indices overlap, if they do, adjust attributes accordingly
+        if utils.overlap(self.min_id, self.max_id, base_min, base_max):
+            warnings.warn(f"Overlapping indices detected. To resolve, indices "
+                          f"where null data present will be shifted forward "
+                          f"when stored in profile: {self.name}")
+
+            # Shift indices (min, max, and all indices in null types index
+            base_min = base_min + self.max_id + 1
+            base_max = base_max + self.max_id + 1
+
+            base_nti = {k: {x + self.max_id + 1 for x in v} for k, v in
+                        base_stats["null_types"].items()}
+
+        # Store/compare min/max id with current
+        if isinstance(base_min, int):
+            if not isinstance(self.min_id, int):
+                self.min_id = base_min
+            else:
+                self.min_id = min(self.min_id, base_min)
+        if isinstance(base_max, int):
+            if not isinstance(self.max_id, int):
+                self.max_id = base_max
+            else:
+                self.max_id = max(self.max_id, base_max)
```

Comment:
I think this can be slightly more clear if we do the following assuming min_id is either `None` or an `int`:
```python
if self.min_id is None:
    self.min_id = base_min
elif base_min is not None and base_min < self.min_id:
    self.min_id = base_min
    
if self.max_id is None:
    self.max_id = base_max
elif base_max is not None and base_max > self.max_id:
    self.max_id = base_min
```
thoughts? not going to block the pr on this though.

Reply:
or could do:
```python
if self.min_id is None:
    self.min_id = base_min
elif base_min is not None:
    self.min_id = min(self.min_id, base_min)
    
if self.max_id is None:
    self.max_id = base_max
elif base_max is not None:
    self.max_id = max(self.max_id, base_max)
```

Reply:
If we do this refactor, we might want to make `self.min_id` -> `self._min_id` and `self.max_id` -> `self._max_id` bc user shouldn't really be interacting with these.

Reply:
Done